### PR TITLE
Small noop changes to Spack package

### PR DESCRIPTION
**Pull request by white238**
Not rebuilding as they have no changes to our TPLs due to forced versions on our end.

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

**Review by kanye-quest — APPROVED** [review]
👍

**Comment by white238** [comment]
I'll merge once the Spack PR gets approved and verify there arent any other required changes.

https://github.com/spack/spack/pull/28067

**Review by agcapps — APPROVED** [review]
👍

**Review by cyrush — APPROVED** [review]

File: scripts/spack/packages/axom/scr_examples_gtest.patch
Comment (white238) [inline scripts/spack/packages/axom/scr_examples_gtest.patch:1]: Spack's CI found a failure in our SPIO examples file.  Basically gtest was being added regardless of ENABLE_TESTS/AXOM_ENABLE_TESTS.  So `axom~tests+examples` woud fail.  This patches the last two releases in Spack.

File: src/axom/sidre/examples/spio/CMakeLists.txt
Comment (white238) [inline src/axom/sidre/examples/spio/CMakeLists.txt:44]: This example wasnt added anywhere and was the only example to need gtest.

**Comment by white238** [comment]
@kennyweiss @agcapps and/or @kanye-quest could you please re-review this? I had to fix a couple things that the Spack PR exposed and also added guards against building gtest/gmock when tests are disabled by the user.

**Review by cyrush — APPROVED** [review]

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

**Review by agcapps — APPROVED** [review]
Thanks for your thoroughness, @white238

**Comment by agcapps** [comment]
> @kennyweiss @agcapps and/or @kanye-quest could you please re-review this? I had to fix a couple things that the Spack PR exposed and also added guards against building gtest/gmock when tests are disabled by the user.

Done.  :thumbsup:

**Pull request merged**
